jetstream/tests/integration/test_config_integration.py

```diff
@@ -228,3 +231,72 @@ def test_busted_default_config_fails(self):
         )
         with pytest.raises(DryRunFailedError):
             validate(extern)
+
+    def test_linear_models_covariate_parsing(self):
+        config = dedent(
+            """\
+            [metrics]
+            weekly = ["bogus_metric"]
+
+            [metrics.bogus_metric]
+            select_expression = "SUM(fake_column)"
+            data_source = "source_name"
+
+            [metrics.bogus_metric.statistics.linear_model_mean]
+            [metrics.bogus_metric.statistics.linear_model_mean.covariate_adjustment]
+            metric = "bogus_metric"
+            period = "preenrollment_week"
```

Comment:
What happens when period is configured here as `weekly`? Does metric-config-parser need to flag that as an error in CI if it gets put into a PR on metric-hub? Are there other config parameter combinations that we want to check for that might be individually valid but invalid in combination with others?

Reply:
As per discussion in slack, we're not going to allow this. Only preenrollment values are valid. We will add runtime checks here to Jetstream, but should add parse-time checks to metric-config-parser as well.

Reply:
Just kidding: metric-config-parser doesn't have the ability to validate individual statistics. I've added a validator to the field in the `LinearModelMean` statistic, so it will be validated at the time that the parsed config is converted from a metric-config-parser Statistic to a Jetstream Statistic.

@mikewilli: can we add a step to the metric-hub CI to validate the statistics? I don't currently believe that Statistic validation is done there.

Reply:
Seems like a good idea. For now I created a ticket, and will try to take a look at it this week https://github.com/mozilla/metric-hub/issues/408